storage: many tests failed

Many tests failed at the same time in this package.

* I will close this issue when there are no more failures in this package _and_
  there is at least one pass.
* No new issues will be filed for this package until this issue is closed.
* If there are already issues for individual test cases, I will close them when
  the corresponding test passes. You can close them earlier, if you prefer, and
  I won't reopen them while this issue is still open.

Here are the tests that failed:
* TestBucketAttrsToRawBucket (#10333)
* TestBucketAttrsToUpdateToRawBucket (#10175)
* TestBucketAttrsToProtoBucket (#10176)
* TestCopyMissingFields (#10177)
* TestSetHeadersFromContext
* TestIntegration_ZeroSizedObject (#10334)
* TestIntegration_NonexistentObject
* TestIntegration_NonexistentBucket (#10336)
* TestIntegration_NoUnicodeNormalization (#10337)
* TestInvoke
* TestShouldRetry
* TestPostPolicyV4OptionsClone
* TestV2HeaderSanitization
* TestV4HeaderSanitization
* TestObjectNames
* TestConditionErrors
* TestObjectCompose
* TestEmptyObjectIterator
* TestEmptyBucketIterator
* TestCodecUint32
* TestUserProject
* TestRawObjectToObjectAttrs
* TestObjectAttrsToRawObject
* TestProtoObjectToObjectAttrs
* TestObjectAttrsToProtoObject
* TestAttrToFieldMapCoverage
* TestSignedURLOptionsClone
* TestTraceSpansMultiEmulated/http/Bucket.Attrs


-----
commit: 48022acb5550244ffdff9a87f2792731eec9b005
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/620b3e7b-e536-4fcc-86d9-6ab7e5ba68c6), [Sponge](http://sponge2/620b3e7b-e536-4fcc-86d9-6ab7e5ba68c6)
status: failed

I believe it's just one test that is failing with a panic causing paused tests to fail: 
```
=== RUN   TestTraceSpansMultiEmulated/http/Bucket.Attrs
--- FAIL: TestTraceSpansMultiEmulated (0.07s)
    --- FAIL: TestTraceSpansMultiEmulated/http (0.00s)
        --- FAIL: TestTraceSpansMultiEmulated/http/Bucket.Attrs (0.00s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1ac17bf]

goroutine 12730 [running]:
testing.tRunner.func1.2({0x1d1e140, 0x2f90f50})
	/[usr/local/go/src/testing/testing.go:1631](https://cs.corp.google.com/piper///depot/google3/usr/local/go/src/testing/testing.go?l=1631) +0x3f7
testing.tRunner.func1()
	/[usr/local/go/src/testing/testing.go:1634](https://cs.corp.google.com/piper///depot/google3/usr/local/go/src/testing/testing.go?l=1634) +0x6b6
panic({0x1d1e140?, 0x2f90f50?})
	/[usr/local/go/src/runtime/panic.go:770](https://cs.corp.google.com/piper///depot/google3/usr/local/go/src/runtime/panic.go?l=770) +0x132
[cloud.google.com/go/storage.(*Client).Bucket(..](http://cloud.google.com/go/storage.(*Client).Bucket(..).)
	/tmpfs/src/google-cloud-[go/storage/bucket.go:62](http://goto.google.com/storage/bucket.go:62)
[cloud.google.com/go/storage.(*emulatedTraceTest).populateResources(0xc034327ce0](http://cloud.google.com/go/storage.(*emulatedTraceTest).populateResources(0xc034327ce0), {0x22b78f8, 0xc0024dff10}, 0x0, {0xc032a20a10, 0x1, 0x1?})
	/tmpfs/src/google-cloud-[go/storage/trace_test.go:228](http://goto.google.com/storage/trace_test.go:228) +0x11f
[cloud.google.com/go/storage.TestTraceSpansMultiEmulated.func1.8(0xc002039520)](http://cloud.google.com/go/storage.TestTraceSpansMultiEmulated.func1.8(0xc002039520))
	/tmpfs/src/google-cloud-[go/storage/trace_test.go:154](http://goto.google.com/storage/trace_test.go:154) +0xf3
```

--- a/storage/trace_test.go
+++ b/storage/trace_test.go
@@ -78,6 +78,7 @@ func TestTraceStorageTraceStartEndSpan(t *testing.T) {
 }
 
 func TestTraceSpansMultiEmulated(t *testing.T) {
+	checkEmulatorEnvironment(t)
 	ctx := skipJSONReads(context.Background(), "no reads in test")
 	// To only test storage library layer instrumentation,
 	// we disable transport layer traces for testing purposes.

